WiinUPro/Controls/NintyControl.xaml.cs

```diff
@@ -394,6 +394,32 @@ public void LoadProfile(string fileName)
 
             if (loadedProfile != null)
             {
+                if (fileName.Length > 6 & fileName[fileName.Length - 6] == '_')
+                {
+                    int n = (int)Char.GetNumericValue(fileName[fileName.Length - 5]);
+                    bool changeProfile = false;
+                    if (n == 1 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_A))
+                    {
+                        changeProfile = true;
+                    }
+                    else if (n == 2 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_B))
+                    {
+                        changeProfile = true;
+                    }
+                    else if (n == 3 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_C))
+                    {
+                        changeProfile = true;
+                    }
+                    else if (n == 4 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_D))
+                    {
+                        changeProfile = true;
+                    }
+                    if (changeProfile && !_rumbleSubscriptions[n - 1])

```

Comment:
If it were not for the extension specific profiles merely checking whether or not a profile has already been loaded for the controller would suffice, since the check would only have to be performed once, when the first profile is loaded. The existence of extension specific profiles necessitates that the rumble controller corresponds with the number of the profile. This could probably be avoided somehow, but I personally have no use case where I would assign an extension's rumble controller to a different one than the its corresponding XInput device.